GPU/GLES/DrawEngineGLES.cpp

```diff
@@ -272,36 +272,49 @@ void DrawEngineGLES::DoFlush() {
 	GLRBuffer *vertexBuffer = nullptr;
 	GLRBuffer *indexBuffer = nullptr;
 	uint32_t vertexBufferOffset = 0;
-	uint32_t indexBufferOffset = 0;
+	// indexBufferOffset is preserved between draws.
 
 	if (vshader->UseHWTransform()) {
-		int vertexCount = 0;
-		bool useElements = true;
 		bool populateCache = false;
+		bool reuseVertexData = false;
 
 		if (g_Config.bSoftwareSkinning && (lastVType_ & GE_VTYPE_WEIGHT_MASK)) {
 			// If software skinning, we've already predecoded into "decoded". So push that content.
 			size_t size = decodedVerts_ * dec_->GetDecVtxFmt().stride;
 			u8 *dest = (u8 *)frameData.pushVertex->Push(size, &vertexBufferOffset, &vertexBuffer);
 			memcpy(dest, decoded, size);
 		} else {
-			// Decode directly into the pushbuffer
-			u8 *dest = (u8 *)DecodeVertsToPushBuffer(frameData.pushVertex, &vertexBufferOffset, &vertexBuffer);
+			if (dcid_ == prevDcid_) {
```

Comment:
Could we still guard this on some ini setting, even if it's not exposed?  I'd worry that we run into a case where this hash is wrong.  It'd be nice to be able to troubleshoot without a new build.

-[Unknown]

Reply:
Yeah, might be a good idea.